libcoap/src/message/response.rs

```diff
@@ -142,6 +142,26 @@ impl CoapResponse {
         self.etag = etag
     }
 
+    /// Returns the "Echo" option value for this request.
+    pub fn echo(&self) -> Option<&Echo> {
+        self.echo.as_ref()
+    }
+
+    /// Sets the "Echo" option value for this response.
+    ///
+    /// This option can be used by servers to ensure that a request is recent.
+    ///
+    /// The client should include the provided request in its response.
```

Comment:
I think this line needs to be slightly adjusted, right?

```suggestion
    /// The client should include the provided option value in its next request.
```